service/src/main/java/org/ehrbase/service/FolderServiceImp.java

```diff
@@ -108,20 +117,175 @@ public ObjectVersionId create(UUID ehrId, Folder content, UUID contribution, UUI
             description = contributionAccess.getAuditsDescription();
         }
 
+        if (systemId == null || committerId == null) {
+            throw new InternalServerException("Error on contribution handling for folder creation.");
+        }
+
         // Get first FolderAccess instance
         I_FolderAccess folderAccess = FolderAccess.buildNewFolderAccessHierarchy(
-                getDataAccess(),
-                content,
-                currentTimeStamp,
-                ehrId,
-                contributionAccess
+            getDataAccess(),
+            objData,
+            currentTimeStamp,
+            ehrId,
+            contributionAccess
         );
-        ObjectVersionId folderId = folderAccess.create(contribution, systemId, committerId, description);
+        ObjectVersionId folderId;
+        if (contribution == null) {
+            folderId = new ObjectVersionId(
+                folderAccess.commit(LocalDateTime.now(), systemId, committerId, description).toString()
+                    + "::" + getServerConfig().getNodename()
+                    + "::1"
+            );
+        } else {
+            folderId = new ObjectVersionId((folderAccess.commit(LocalDateTime.now(), contribution).toString()
+                + "::" + getServerConfig().getNodename()
```

Comment:
@jakesmolka Are we are overwriting the values sent by the client for the object version id? For instance, for the creating_system_id what happens if the client sends something different than the value we have configured? https://specifications.openehr.org/releases/ITS-REST/latest/ehr.html#contribution-contribution-post

Reply:
Interesting question again! But this is kind of out of scope of this PR. Anyway, here are my thoughts on that:

In EHRbase, if a client commits an object via direct or contribution endpoint, the given UID will be ignored and handled server-side.

I'm talking about what's illustrated in that little composition snippet:

``` json
{
  "_type": "COMPOSITION",
  ...
  "uid": {
    "_type": "OBJECT_VERSION_ID",
    "value": "8849182c-82ad-4088-a07f-48ead4180515::openEHRSys.example.com::1"
  },
  ...
```

Overall there's some discussion in the SEC, because this behavior is not really 100% clear in the API docs. See: https://openehr.atlassian.net/browse/SPECITS-62 
Please comment your perspective over there :slightly_smiling_face: 

And specifically regarding the "creating_system_id": EHRbase doesn't support imported versions or alike yet. So there's no concept of data from other systems. Data committed (especially like in this case via initial object creation) will be treated to be originally created in the local EHRbase instance. Therefore it always used the current internal system ID.